### DISCUSSION TITLE
Reload add-on in case of failover

### DISCUSSION THREAD

**Discussion by tristangrichard** [homeassistant-apps/app-cloudflared#512] [Ideas]
Hi,

First of all, thanks for this great add-on!
I have a mobile router acting as a failover WAN for when my primary WAN fails. Could the add-on detect the external IP switch and notify the CloudFlare services so the tunnel can be redirected?

**Comment by brenner-tobias** [comment]
This should already work, though I am not sure if and how long the delay might be. The solution is not just forwarding packages to some ip, but the tunnel, which is started from your client (HA) is talking to the Cloudflare Servers. Therefore, it does not matter which IP this is coming from.
Since I am curious though: It would be great if you could test, if and how fast this is working after unplugging your primary WAN.

**Reply by tristangrichard** [reply]
Sorry for the late reply. Must have skipped the notification of your comment.
You are right, It works and updates within few seconds. 
Must have been something with the failover connection
